### ISSUE TITLE
--recursive to test an entire workspace?

### ISSUE THREAD

**Issue by matthiaskrgr** [nextest-rs/nextest#36]
Hi, first of all awesome crate! :D
On just a dual core system it seems to be 10-15% faster than default `cargo test`. :) 

While playing around with it, I wondered if it would be useful to have some kind of `--recursive` flag that does not only run `test` in the current directory but also on all other crates of the current workspace.

**Comment by epage** [comment]
Isn't this handed via `--workspace`?

**Comment by sunshowers** [comment]
Thanks for the report! I'm glad that even a dual core system is faster.

As @epage says, this is handled by `--workspace`. nextest just forwards the build command to `cargo test --no-run`, so the package selection is identical to `cargo test`.

**Comment by matthiaskrgr** [comment]
Hmm.
I was playing around in the clippy repo.
When I `cd clippy_lints ; cargo test`, it runs tests and doctests that are not executed with either `cargo test` or `cargo test --workspace`, I guess this means that this crate that happens to be in the same git repo is not part of the workspace..?

The idea was to have `--recursive` run these tests as well.

It also looks like `cargo nextest run` skips the doctests?

**Comment by sunshowers** [comment]
Yes, unfortunately doctests aren't yet supported because of limitations in stable Rust. Going to mark this as closed for now as a result. Thanks again for reporting the issue!

**Comment by matthiaskrgr** [comment]
Well I would like `--recursive` to do something like:
find all projects (`cargo.toml`s) that are either in the current workspace or in a subdirectory,
run `cargo nextest` on all of these (in parallel).

This does not seem to be what `--workspace` does because it requires that I actually have configured a workspace in my `Cargo.toml`.
`--recursive` should just find all the crates on its own imo (if that's possible) :)
